quote-server/server.js

```diff
@@ -1,32 +1,59 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
-import quotes from "./quotes.json" assert { type: "json" };
+import crypto from "crypto";
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
-app.get("/", (request, response) => {
-  response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
+
+let quotes = [];
+
+app.use(cors());
+
+app.use(async (req, res, next) => {
+  const response = await fetch("https://api.quotable.io/quotes?limit=150");
+  const data = await response.json();
+  quotes = data.results;
+  next();
+});
+
+app.get("/", (req, res) => {
+  res.send(
+    "Fikret's Quote Server! Ask me for /quotes, /quotes/random or /quotes/search?term=(something)"
+  );
 });
 
 //START OF YOUR CODE...
+app.get("/quotes", (req, res) => {
+  res.json({ quotes });
+});
 
+app.get("/quotes/random", (req, res) => {
+  const randomQuote = pickFromArray(quotes);
+  res.json({ randomQuote });
+});
+
+app.get("/quotes/search", (req, res) => {
+  console.log(req.query);
+  const searchTerm = req.query.term.toLowerCase();
```

Comment:
Nice extracting of a variable here to avoid re-doing the `toLowerCase()` call for each quote